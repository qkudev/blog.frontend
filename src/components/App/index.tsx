import * as React from 'react'
import { Page, Layout } from '../index'
import { Routes } from '../../pages'
import './style'

class App extends React.Component {
  render() {
    return <Routes />
  }
}

export default App
