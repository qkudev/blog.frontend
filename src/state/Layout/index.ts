import * as Actions from './actions'

export * from './reducer'
export { Actions }
