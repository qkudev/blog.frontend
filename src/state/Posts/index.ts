import * as Actions from './actions'

export { Actions }
export * from './reducer'
