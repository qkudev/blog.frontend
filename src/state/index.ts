import { applyMiddleware, combineReducers, createStore, Middleware } from 'redux'
import thunk from 'redux-thunk'
import { composeWithDevTools } from 'redux-devtools-extension'
import logger from 'redux-logger'

const isEnvProduction = process.env.NODE_ENV === 'production'

let middlewares: Middleware[] = [thunk]

if (!isEnvProduction) {
  middlewares.push(logger)
}

export interface IState {
}

export const initState: IState = {}

const rootReducer = combineReducers({})

export const store = createStore(
  rootReducer,
  initState,
  isEnvProduction
    ? applyMiddleware(...middlewares)
    : composeWithDevTools(applyMiddleware(...middlewares))
)
