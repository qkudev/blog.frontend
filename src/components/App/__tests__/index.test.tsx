import * as React from 'react'
import * as ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import App from '../index'
import * as State from '../../../state'
import { MemoryRouter } from 'react-router-dom'

describe('App', function() {
  it('should render without crashing', function() {
    const div = document.createElement('div')
    ReactDOM.render(
      <State.Provider store={State.store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </State.Provider>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches with snapshot', function() {
    const app = renderer.create(
      <State.Provider store={State.store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </State.Provider>
    )
    let tree = app.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
