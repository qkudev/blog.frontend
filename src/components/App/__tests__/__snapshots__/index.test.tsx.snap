// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Matches with snapshot 1`] = `
<div
  className="page "
>
  <div
    className="app"
  >
    <div
      className="layout header"
      id="app-header"
    >
      <a
        href="/"
        id="header-link"
        onClick={[Function]}
      >
        Show me them all
      </a>
    </div>
    <div
      className="layout content"
      id="content"
    >
      <div>
        <div
          className="lds-facebook"
        >
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
